server/src/test/scala/org/http4s/server/middleware/authentication/AuthMiddlewareSpec.scala

```diff
@@ -0,0 +1,48 @@
+package org.http4s.server.middleware.authentication
+
+import cats.data.Kleisli
+import fs2.Task
+import org.http4s.server.AuthMiddleware
+import org.http4s._
+import org.http4s.dsl._
+
+class AuthMiddlewareSpec extends Http4sSpec {
+
+  "AuthMiddleware" should {
+    "work" in {
```

Comment:
Yes, agreed. This was meant as an absolute first draft. I'll get back to all your comments soon. 
Could you please let me know if the typeclass should be placed somewhere else? That's my main problem at the moment.